pkg/lease/manager.go

```diff
@@ -45,6 +45,15 @@ type manager struct {
 	log            logr.Logger
 }
 
+// AlreadyHeldError is returned in case the lease that is requested is already held by a different holder
+type AlreadyHeldError struct {
+	holderIdentity string
+}
+
+func (e *AlreadyHeldError) Error() string {
```

Comment:
missed in the review... error types should have the Error() method on the instance receiver, not pointer. Only saw it now  in the NHC PR because `_, isLeaseAlreadyTaken := err.(*lease.AlreadyHeldError)` looks really unfamiliar 🙈 Do you mind to change?